transport/src/main/java/io/netty/channel/pool/FixedChannelPool.java

```diff
@@ -0,0 +1,359 @@
+     * @param maxConnections        the numnber of maximal active connections, once this is reached new tries to
+     *                              acquire a {@link Channel} will be delayed until a connection is returned to the
+     *                              pool again.
+     * @param maxPendingAcquires    the maximum number of pending acquires. Once this is exceed acquire tries will
+     *                              be failed.
+     */
+    public FixedChannelPool(Bootstrap bootstrap,
+                            ChannelPoolHandler  handler,
+                            ChannelHealthChecker healthCheck, AcquireTimeoutAction action,
+                            final long acquireTimeoutMillis,
+                            int maxConnections, int maxPendingAcquires) {
+        super(bootstrap, handler, healthCheck);
+        if (maxConnections < 1) {
+            throw new IllegalArgumentException("maxConnections: " + maxConnections + " (expected: >= 1)");
+        }
+        if (maxPendingAcquires < 1) {
+            throw new IllegalArgumentException("maxPendingAcquires: " + maxPendingAcquires + " (expected: >= 1)");
+        }
+        if (action == null && acquireTimeoutMillis == -1) {
+            timeoutTask = null;
+            acquireTimeoutNanos = -1;
+        } else if (action == null && acquireTimeoutMillis != -1) {
+            throw new NullPointerException("action");
+        } else if (action != null && acquireTimeoutMillis < 0) {
+            throw new IllegalArgumentException("acquireTimeoutMillis: " + acquireTimeoutMillis + " (expected: >= 1)");
+        } else {
+            acquireTimeoutNanos = TimeUnit.MILLISECONDS.toNanos(acquireTimeoutMillis);
+            switch (action) {
+            case Fail:
+                timeoutTask = new TimeoutTask() {
+                    @Override
+                    public void onTimeout(AcquireTask task) {
+                        // Fail the promise as we timed out.
+                        task.promise.setFailure(TIMEOUT_EXCEPTION);
+                    }
+                };
+                break;
+            case NewConnection:
+                timeoutTask = new TimeoutTask() {
+                    @Override
+                    public void onTimeout(AcquireTask task) {
+                        // Increment the acquire count and delegate to super to actually acquire a Channel which will
+                        // create a new connetion.
+                        ++acquiredChannelCount;
+
+                        FixedChannelPool.super.acquire(task.promise);
+                    }
+                };
+                break;
+            default:
+                throw new Error();
+            }
+        }
+        executor = bootstrap.group().next();
+        this.maxConnections = maxConnections;
+        this.maxPendingAcquires = maxPendingAcquires;
+    }
+
+    // We substract here to guard against overflow.
+    private static long timeSinceClassInit() {
```

Comment:
Shouldn't we use the time comparison technique recommended in the [nanoTime java doc](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()) (`t1 - t0 < threshold` instead of `t1 < t0`) instead of introducing this method?